components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RegistryMetadataResource.java

```diff
@@ -76,11 +76,11 @@ public boolean invoke(MessageContext messageContext, org.apache.axis2.context.Me
      *
      * @param axis2MessageContext AXIS2 message context
      */
-    private void handleGet(org.apache.axis2.context.MessageContext axis2MessageContext,
+    private void handleGet(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
             String registryPath, String validatedPath) {
 
         MicroIntegratorRegistry microIntegratorRegistry = new MicroIntegratorRegistry();
```

Comment:
Shall we change `MicroIntegratorRegistry microIntegratorRegistry = new MicroIntegratorRegistry();` to

```
MicroIntegratorRegistry microIntegratorRegistry =
                (MicroIntegratorRegistry) messageContext.getConfiguration().getRegistry();
```